test/e2e/utils.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	capiv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+// IsNodeReady returns true if a node is ready; false otherwise.
+func IsNodeReady(node *corev1.Node) bool {
```

Comment:
Is there a reason why we can't use the canonical implementation in: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/node/util.go